Dockerfile

```diff
@@ -16,20 +16,28 @@
 # limitations under the License.
 #
 
-FROM centos:7
+# Uses intermediate image for building Drill to reduce target image size
+FROM maven:3.6-jdk-8 as build
 
-# Project version defined in pom.xml is passed as an argument
-ARG VERSION
+# Copy project sources into the container
+COPY . /src
 
-# JDK 8 is a pre-requisite to run Drill ; 'which' package is needed for drill-config.sh
-RUN yum install -y java-1.8.0-openjdk-devel which ; yum clean all ; rm -rf /var/cache/yum
+WORKDIR /src
 
-# The drill tarball is generated upon building the Drill project
-COPY target/apache-drill-$VERSION.tar.gz /tmp
+# Builds Drill
+RUN  mvn clean install -DskipTests -q
```

Comment:
Do we want to build Drill in the container? Would seem more reliable/repeatable to use a fixed release.

In my experience, we often need to tweak Dockerfiles for specific needs. In that case, reusing the same Drill image on each build, rather than building a new image, makes the process more stable. Plus, we don't end up with an entire build in the Docker file system.

We could have a separate development image if the purpose of this is to allow that use case.

Reply:
Thanks, moved the previous Dockerfile back, so it may be used as it was done before, but also left new Dockerfile in the project root, so it will be used by DockerHub Automated Build.

Reply:
Does Dockerhub want to do its own build? In the container? Is the purpose of this to do builds using Docker? Or, to build the "official" Drill container? If the latter, I do think we want to use a release of the software and not rebuild. Else, there is no guarantee that two builds of the container image itself have exactly the same Drill bits.

Reply:
Yes, the purpose is to build the "official" Drill container for the releases and for the master branch. There are a lot of other Apache projects which do the same - build project when building the image.

Automating the publishing simplifies things a lot, to publish Drill image, without it, you should create INFRA Jira to provide push access for Drill DockerHub, and only later push the image.

I think that there wouldn't be significant differences between Drill built during the release and Drill, built inside Docker image.

Reply:
Confirm that this is standard practice to have animated Docker builds for Apache projects.
In our case official build will be pushed each time we create release tag. Plus latest (master) build will be pushed with each new commit.